generic-factory/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+        access_keys: Vec<AccessKey>,
+        method_name: Option<String>,
+        args: Option<Base64VecU8>,
+    ) {
+        internal_create(
+            &format!("{}.{}", name, env::current_account_id()),
+            &CryptoHash::from(hash),
+            access_keys,
+            method_name,
+            args.map(|args| args.0),
+        );
+    }
+}
+
+/// Stores attached data into blob store and returns hash of it.
+/// Implemented to avoid loading the data into WASM for optimal gas usage.
+#[cfg(target_arch = "wasm32")]
+#[no_mangle]
+pub extern "C" fn store() {
+    env::setup_panic_hook();
+    env::set_blockchain_interface(Box::new(near_blockchain::NearBlockchain {}));
+    unsafe {
+        BLOCKCHAIN_INTERFACE.with(|b| {
+            // Load input into register 0.
+            b.borrow()
+                .as_ref()
+                .expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR)
+                .input(0);
+            // Compute sha256 hash of register 0 and store in 1.
+            b.borrow()
+                .as_ref()
+                .expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR)
+                .sha256(u64::MAX as _, 0 as _, 1);
+            // Check if such blob already stored.
+            assert_eq!(
+                b.borrow()
+                    .as_ref()
+                    .expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR)
+                    .storage_has_key(u64::MAX as _, 1 as _),
+                0,
+                "ERR_ALREADY_EXISTS"
+            );
+            // Get length of the input argument and check that enough $NEAR has been attached.
+            let blob_len = b
+                .borrow()
+                .as_ref()
+                .expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR)
+                .register_len(0);
+            let storage_cost = ((blob_len + 32) as u128) * env::storage_byte_cost();
+            assert!(
+                env::attached_deposit() >= storage_cost,
+                "ERR_NOT_ENOUGH_DEPOSIT:{}",
+                storage_cost
+            );
+            // Store value of register 0 into key = register 1.
+            b.borrow()
+                .as_ref()
+                .expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR)
+                .storage_write(u64::MAX as _, 1 as _, u64::MAX as _, 0 as _, 2);
+            // Load register 1 into blob_hash and save into LookupMap.
+            let blob_hash = [0u8; 32];
+            b.borrow()
+                .as_ref()
+                .expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR)
+                .read_register(1, blob_hash.as_ptr() as _);
+            // Return from function value of register 1.
+            let blob_hash_str = near_sdk::serde_json::to_string(&Base58CryptoHash::from(blob_hash))
+                .unwrap()
+                .into_bytes();
+            b.borrow()
+                .as_ref()
+                .expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR)
+                .value_return(blob_hash_str.len() as _, blob_hash_str.as_ptr() as _);
+        });
+    }
```

Comment:
```suggestion
    BLOCKCHAIN_INTERFACE.with(|b| unsafe {
        let interface = b.borrow().as_ref().expect(BLOCKCHAIN_INTERFACE_NOT_SET_ERR);
        // Load input into register 0.
        interface.input(0);
        // Compute sha256 hash of register 0 and store in 1.
        interface.sha256(u64::MAX as _, 0 as _, 1);
        // Check if such blob already stored.
        assert_eq!(
            interface.storage_has_key(u64::MAX as _, 1 as _),
            0,
            "ERR_ALREADY_EXISTS"
        );
        // Get length of the input argument and check that enough $NEAR has been attached.
        let blob_len = interface.register_len(0);
        let storage_cost = ((blob_len + 32) as u128) * env::storage_byte_cost();
        assert!(
            env::attached_deposit() >= storage_cost,
            "ERR_NOT_ENOUGH_DEPOSIT:{}",
            storage_cost
        );
        // Store value of register 0 into key = register 1.
        interface.storage_write(u64::MAX as _, 1 as _, u64::MAX as _, 0 as _, 2);
        // Load register 1 into blob_hash and save into LookupMap.
        let blob_hash = [0u8; 32];
        interface.read_register(1, blob_hash.as_ptr() as _);
        // Return from function value of register 1.
        let blob_hash_str = near_sdk::serde_json::to_string(&Base58CryptoHash::from(blob_hash))
            .unwrap()
            .into_bytes();
        interface.value_return(blob_hash_str.len() as _, blob_hash_str.as_ptr() as _);
    });
```

You can simplify these calls so it's a bit easier to read if you want, and it will make things easier to migrate to the new interface when available.

I can provide suggestions for the other usages if you'd like :)